cmake/Modules/Packages/GPU.cmake

```diff
@@ -99,6 +99,9 @@ if(GPU_API STREQUAL "CUDA")
   add_library(gpu STATIC ${GPU_LIB_SOURCES} ${GPU_LIB_CUDPP_SOURCES} ${GPU_OBJS})
   target_link_libraries(gpu PRIVATE ${CUDA_LIBRARIES} ${CUDA_CUDA_LIBRARY})
   target_include_directories(gpu PRIVATE ${LAMMPS_LIB_BINARY_DIR}/gpu ${CUDA_INCLUDE_DIRS})
+  if(NOT BUILD_MPI)
+    target_include_directories(gpu PRIVATE ${LAMMPS_SOURCE_DIR}/STUBS)
+  endif()
```

Comment:
Could you just do `target_link_libraries(gpu PRIVATE MPI::MPI_CXX)` at the very bottom on `GPU.cmake`?

Reply:
that is already there

Reply:
I think here is what is going on and I had similar discussion with @rbberger on #1956 as well. `find_package(MPI)` will create an incomplete `MPI::MPI_CXX` target and hence the aliasing of `mpistubs` to `MPI::MPI_CXX` will not forward the include directories.

One workaround to this would be to not use a target in the `MPI::` namespace, e.g. `LAMMPS::MPI`, however aliasing imported target, like `MPI::MPI_CXX` (from `find_package(MPI)` ) isn't supported in CMake-3.10 yet, so this wasn't an option either.

See https://github.com/akohlmey/lammps/pull/65 for slightly cleaner version of what you had.